src/main/scala/io/iohk/ethereum/domain/branch/EmptyBlockchainBranch.scala

```diff
@@ -0,0 +1,11 @@
+package io.iohk.ethereum.domain.branch
+import akka.util.ByteString
+
+import io.iohk.ethereum.domain.Block
+
+object EmptyBlockchainBranch extends BlockchainBranch {
```

Comment:
I wonder if the empty branch is possible, since we always have at least genesis block

Reply:
I completely agree. The `EmptyBlockchainBranch` implementation is a compromise to allow refactoring without breaking everything. 

The current implementation does not guarantee that a genesis is set (it must be done by a side effect somewhere else), so lot of unit tests would fail when trying to get the best branch when no genesis was set. My previous implementation returned an `Option[BlockchainBranch]` to handle this, but this was tedious to use for no real reason. This implementation allows to keep a nicer interface without breaking the tests.